compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2712,11 +2715,35 @@ public void visit(BLangQueryExpr queryExpr) {
                         this.dlog.error(collection.pos, DiagnosticCode.NOT_ALLOWED_STREAM_USAGE_WITH_FROM);
                     }
                 }
+            } else if (clause.getKind() == NodeKind.ORDER_BY) {
+                checkOrderFieldValidity(queryExpr.type, (BLangOrderByClause) clause);
             }
             analyzeNode(clause, env);
         }
     }
 
+    private void checkOrderFieldValidity(BType resultType, BLangOrderByClause clause) {
+        resultType = types.resolveExprType(resultType);
+        if (resultType.tag != TypeTags.RECORD) {
+            return;
+        }
+        BRecordType recordType = (BRecordType) resultType;
+        Map<String, BField> recordFields = recordType.fields;
+        for (OrderKeyNode orderKeyNode : clause.getOrderKeyList()) {
+            if (!recordFields.containsKey(orderKeyNode.getOrderKey().toString())) {
```

Comment:
You can move orderKeyNode.getOrderKey().toString() as common local var

Reply:
This was removed in a later PR (https://github.com/ballerina-platform/ballerina-lang/pull/25108)